synapse/lib/storm.py

```diff
@@ -2123,14 +2127,14 @@ async def getInput(self):
             if len(buid) != 32:
                 raise s_exc.NoSuchIden(mesg='Iden must be 32 bytes', iden=iden)
 
-            node = await self.snap.getNodeByBuid(buid)
+            node = await self.view.getNodeByBuid(buid)
             if node is not None:
                 yield node, self.initPath(node)
 
     def layerConfirm(self, perms):
         if self.asroot:
             return
-        iden = self.snap.wlyr.iden
+        iden = self.view.layers[0].iden
```

Comment:
I've seen this pattern in a few places. It's trading two property refs for two property refs and a index lookup.

Would it be bad to have a `view.wlyr` property that points to the write layer of the View?
We can set that when we compute view layers; so that re-setting the layers of a View would set the correct `wlyr` property.

Reply:
I think that would be fine, I had started switching it at one point but left it for now, I'll go back and switch it over.